### PR TITLE
Hosting test timeouts

### PR DESCRIPTION
**Pull request by JamesNK**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Checklist

- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [ ] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6387)

### CODE REVIEW

**Comment by mitchdenny** [comment]
I think merging this is a good idea. The only question is about timing (in terms of when to merge AND individual timeouts).

Some of our test cases need to be modified to use time provider but that will take some time to do that so some tests might become flakey initially whilst that gets worked through. So should we wait until 9.0 has a stable build.

(acknowledging that we are currently being impacted by flakey tests).

File: tests/Aspire.StackExchange.Redis.Tests/RedisContainerFixture.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public async Task DisposeAsync()
     public static async Task<RedisContainer> CreateContainerAsync()
     {
         var container = new RedisBuilder()
-            .WithImage($"{TestConstants.AspireTestContainerRegistry}/{RedisContainerImageTags.Image}:{RedisContainerImageTags.Tag}")
+            .WithImage($"{ComponentTestConstants.AspireTestContainerRegistry}/{RedisContainerImageTags.Image}:{RedisContainerImageTags.Tag}")
```
Comment (mitchdenny) [inline tests/Aspire.StackExchange.Redis.Tests/RedisContainerFixture.cs:37]: Unrelated?

Comment (JamesNK) [reply tests/Aspire.StackExchange.Redis.Tests/RedisContainerFixture.cs:37]: I added a TestConstants type and renamed this one to avoid conflict.

**Comment by JamesNK** [comment]
> Some of our test cases need to be modified to use time provider but that will take some time to do that so some tests might become flakey initially whilst that gets worked through. So should we wait until 9.0 has a stable build.

Do you want this to be in the 9.0 branch? While these are just test changes, there are a lot of test changes. There are benefits: it would make backporting test changes easier. Could make 9.0 branch less flaky.

I was imagining these changes would be merged on main and not be backported.

**Comment by JamesNK** [comment]
I don't want this PR to rot. If makes changes all over the place and will gather conflicts easily.

What feedback do you have? What changes are required before it can be merged?

**Comment by mitchdenny** [comment]
If you are suggesting backporting this to release/9.0 I am OK with it ... but we should make sure we can get all the way to release/9.0 for sure before merging this to main otherwise every other backport will become painful because of the number of source files this change touches.

Fact that this is already starting to accumulate merge conflicts kind of makes my point. I'm 100% supportive of this change, it's just about timing (I like how this gives us localized control over wait behavior vs. some global approach).

**Review by adamint — APPROVED** [review]

**Comment by JamesNK** [comment]
> If you are suggesting backporting this to release/9.0

No

**Comment by mitchdenny** [comment]
I think this is a good change. Its heavy handed but I think test reliability will improve if we can more easily identify where hangs are occuring.

**Review by mitchdenny — APPROVED** [review]
